staff/tests/views/coupon/test_coupon_new_batch.py

```diff
@@ -126,3 +127,15 @@ def test_coupon_generation_creates_correct_coupons(logged_in_existing_user, coup
     assert Coupon.objects.count() == 2
     assert test_coupon_effects == [coupon_effect_basics, coupon_effect_bootcamp]
     assert test_coupon.description == 'test1@test.com'
+
+def test_send_cooupon_email_batch(logged_in_existing_user, mocker, course_basics, course_bootcamp):
+    mocker.patch('emails.library.sendgrid.Sendgrid.send')
+    test_file_path = "./staff/tests/forms/csv_files/correct_test_file.csv"
+    csv_file = open(test_file_path, 'r')
+    content = csv_file.read()
+    uploaded_file = SimpleUploadedFile(name=csv_file.name, content=bytes(content, 'utf-8'), content_type="multipart/form-data")
+
+    client.post(reverse('coupon_new_batch'), {'csv_file': uploaded_file})
+
+    Sendgrid.send.assert_called()
```

Comment:
can use assert_called_with to check that the variables we pass in are correct? and check that the coupons objects created have the attributes we assigned?

Reply:
I can't access / remake the message object we created using the `sendgrid.helper.Mail()`